pkg/core/variationDecider.go

```diff
@@ -68,18 +68,22 @@ func GetVariation(vwoInstance schema.VwoInstance, userID string, campaign schema
 		options.VariationTargetingVariables["_vwo_user_id"] = userID
 	}
 
-	targettedVariation, err := FindTargetedVariation(vwoInstance, userID, campaign, options)
+	targettedVariation, level, err := FindTargetedVariation(vwoInstance, userID, campaign, options)
 	if err != nil {
-		utils.LogMessage(vwoInstance.Logger, constants.Error, variationDecider, err.Error())
-	} else {
+		utils.LogMessage(vwoInstance.Logger, constants.Info, variationDecider, err.Error())
+	}else {
 		message := fmt.Sprintf(constants.InfoMessageGotVariationForUser, vwoInstance.API, userID, campaign.Key, campaign.Type, targettedVariation.Name)
 		utils.LogMessage(vwoInstance.Logger, constants.Info, variationDecider, message)
 		return targettedVariation, nil
 	}
 
-	variationName, err := GetVariationFromUserStorage(vwoInstance, userID, campaign)
+	variationName, level, err := GetVariationFromUserStorage(vwoInstance, userID, campaign)
```

Comment:
why do you need level? can't we do this in the function itself?

Reply:
If we log the issue in the function itself, we wont be able to handle the cases in the calling function because there we are checking if the err := nil

Reply:
Or we can return some dummy error and just handle the case with logging in the function not in the calling function